.github/workflows/ci.yml

```
@@ -4,26 +4,80 @@ on:
   - pull_request
 
 jobs:
-  test:
+  setup_matrix:
+    name: 'Setup Test Matrix'
     runs-on: ubuntu-latest
+    timeout-minutes: 40
+    outputs:
+      beaker_setfiles: ${{ steps.get-outputs.outputs.beaker_setfiles }}
+      puppet_major_versions: ${{ steps.get-outputs.outputs.puppet_major_versions }}
+      puppet_unit_test_matrix: ${{ steps.get-outputs.outputs.puppet_unit_test_matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install the Puppet PDK
+        run: |
+          CODENAME=$(grep UBUNTU_CODENAME /etc/os-release | cut -d '=' -f2)
+          wget https://apt.puppet.com/puppet-tools-release-$CODENAME.deb
+          sudo apt-get update
+          sudo dpkg -i puppet-tools-release-$CODENAME.deb
```

I like that you can source it instead of having to parse it.
```suggestion
          source /etc/os-release
          wget https://apt.puppet.com/puppet-tools-release-${UBUNTU_CODENAME}.deb
          sudo apt-get update
          sudo dpkg -i puppet-tools-release-${UBUNTU_CODENAME}.deb
```

---

I didn't realize you could do that, that is pretty awesome